src/test/java/com/commercetools/sync/commons/utils/GenericUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,264 @@
+                customFieldValue, channel, mock(CategorySyncOptions.class)).orElse(null);
+
+        assertThat(updateAction).isNotNull();
+        assertThat(updateAction.getAction()).isEqualTo("setCustomField");
+        assertThat(updateAction.getClass().getName())
+                .isEqualTo("io.sphere.sdk.channels.commands.updateactions.SetCustomField");
+    }
+
+    @Test
+    public void buildTypedSetCustomFieldUpdateAction_WithNonHandledResource_ShouldNotBuildUpdateAction() {
+        // Cart resource is not handled by buildTypedUpdateAction()
+        final Cart cart = mock(Cart.class);
+        when(cart.getId()).thenReturn("cartId");
+        when(cart.toReference()).thenReturn(Reference.of(Cart.referenceTypeId(), "cartId"));
+
+        final JsonNode customFieldValue = mock(JsonNode.class);
+        final String customFieldName = "name";
+
+        final UpdateAction<Cart> updateAction =
+                buildTypedSetCustomFieldUpdateAction(customFieldName, customFieldValue, cart,
+                        mock(BaseSyncOptions.class)).orElse(null);
+
+        assertThat(updateAction).isNull();
+    }
+
+    @Test
+    public void buildTypedSetCustomFieldUpdateAction_WithNonHandledResource_ShouldCallSyncOptionsCallBack() {
+        // Cart resource is not handled by buildTypedUpdateAction()
+        final Cart cart = mock(Cart.class);
+        when(cart.getId()).thenReturn("cartId");
+        when(cart.toReference()).thenReturn(Reference.of(Cart.referenceTypeId(), "cartId"));
+
+        final JsonNode customFieldValue = mock(JsonNode.class);
+        final String customFieldName = "name";
+        final ArrayList<Object> callBackResponses = new ArrayList<>();
+        final BiConsumer<String, Throwable> updateActionErrorCallBack = (errorMessage, exception) -> {
+            callBackResponses.add(errorMessage);
+            callBackResponses.add(exception);
+        };
+
+        final BaseSyncOptions baseSyncOptions = mock(BaseSyncOptions.class);
+        when(baseSyncOptions.getUpdateActionErrorCallBack()).thenReturn(updateActionErrorCallBack);
+        doCallRealMethod().when(baseSyncOptions).callUpdateActionErrorCallBack(anyString(), any(Throwable.class));
+
+        final UpdateAction<Cart> updateAction =
+                buildTypedSetCustomFieldUpdateAction(customFieldName, customFieldValue, cart,
+                        baseSyncOptions).orElse(null);
+
+        assertThat(updateAction).isNull();
+        assertThat(callBackResponses).hasSize(2);
+        assertThat(callBackResponses.get(0)).isEqualTo("Failed to build 'setCustomField' update action on the custom " +
+                "field with the name 'name' on the cart with id 'cartId'. Reason: Update actions for resource:" +
+                " 'cart' is not implemented.");
+        assertThat((Exception)callBackResponses.get(1)).isInstanceOf(BuildUpdateActionException.class);
+    }
+
+    @Test(expected = BuildUpdateActionException.class)
```

Comment:
since you already depend on _assertJ_ better to use this approach:
http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#exception-assertion